lib/graphql_metrics/instrumentation.rb

```diff
@@ -1,109 +1,40 @@
-    end
-
-    def self.current_time
-      Process.clock_gettime(Process::CLOCK_MONOTONIC)
-    end
-
-    def use(schema_definition)
-      return unless extractor_defines_any_visitors?
-
-      setup_instrumentation(schema_definition)
-    end
-
-    def setup_instrumentation(schema_definition)
-      schema_definition.instrument(:query, self)
-      schema_definition.instrument(:field, self)
-    end
-
-    def extractor
-      @extractor ||= Extractor.new(self)
-    end
+  module Instrumentation
+    module_function
 
     def before_query(query)
-      return unless extractor_defines_any_visitors?
+      return if query.context[GraphQLMetrics::SKIP_GRAPHQL_METRICS_ANALYSIS]
 
-      ns = query.context.namespace(CONTEXT_NAMESPACE)
-      ns[TIMING_CACHE_KEY] = {}
-      ns[START_TIME_KEY] = self.class.current_time
-    rescue StandardError => ex
-      extractor.handle_extraction_exception(ex)
+      query.context.namespace(CONTEXT_NAMESPACE).tap do |ns|
+        ns[GraphQLMetrics::TIMINGS_CAPTURE_ENABLED] = true
+        ns[GraphQLMetrics::INLINE_FIELD_TIMINGS] = {}
+        ns[GraphQLMetrics::LAZY_FIELD_TIMINGS] = {}
+      end
     end
 
     def after_query(query)
-      @query = query
-
-      return unless extractor_defines_any_visitors?
-      return if respond_to?(:skip_extraction?) && skip_extraction?(query)
-      return unless @ctx_namespace = query.context.namespace(CONTEXT_NAMESPACE)
-
-      before_query_extracted(query, query.context) if respond_to?(:before_query_extracted)
-
-      extractor.extract!(query)
-
-      after_query_teardown(query) if respond_to?(:after_query_teardown)
-    rescue StandardError => ex
-      extractor.handle_extraction_exception(ex)
-    ensure
-      @query = nil
-    end
-
-    def instrument(type, field)
-      return field unless respond_to?(:field_extracted) || extractor.respond_to?(:field_extracted)
-      return field if type.introspection?
-
-      old_resolve_proc = field.resolve_proc
-      new_resolve_proc = ->(obj, args, ctx) do
-        start_time = self.class.current_time
-        result = old_resolve_proc.call(obj, args, ctx)
-
-        begin
-          next result if respond_to?(:skip_field_resolution_timing?) &&
-            skip_field_resolution_timing?(query, ctx)
-
-          end_time = self.class.current_time
-
-          ns = ctx.namespace(CONTEXT_NAMESPACE)
-
-          ns[TIMING_CACHE_KEY][ctx.ast_node] ||= []
-          ns[TIMING_CACHE_KEY][ctx.ast_node] << end_time - start_time
-
-          result
-        rescue StandardError => ex
-          extractor.handle_extraction_exception(ex)
-          result
-        end
+      return if query.context[GraphQLMetrics::SKIP_GRAPHQL_METRICS_ANALYSIS]
+
+      query.context.namespace(CONTEXT_NAMESPACE).tap do |ns|
+        query_duration = GraphQLMetrics.current_time_monotonic - ns[GraphQLMetrics::QUERY_START_TIME_MONOTONIC]
+        query_end_time = GraphQLMetrics.current_time
```

Comment:
this isn't really a useful value since we already have the start time *and* the duration?